.github/workflows/build.yml

```diff
@@ -126,37 +126,14 @@ jobs:
       - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b
         with:
           python-version: 3.11
-      - name: Install poetry (latest release)
-        uses: abatilo/actions-poetry@e78f54a89cb052fff327414dd9ff010b5d2b4dbd
+      - name: Install uv
+        uses: astral-sh/setup-uv@2e657c127d5b1635d5a8e3fa40e0ac50a5bf6992
         with:
-          poetry-version: latest
+          version: "0.5.1"
```

Comment:
@ilyagr I haven't found a good way to mark the job as optional -- see https://github.com/actions/runner/issues/2347 -- so I didn't add an unpinned job.

Reply:
Whether a job is optional or not is determined in repo settings that only Martin can change (AFAIK). Optional is the default. It's based on the name of the job, I'm pretty sure, so you can make the name the same as an existing mandatory job when it makes sense.

Reply:
What I was thinking here is to have the `mkdocs` job you made and then have an extra job:

```yaml
  mkdocs-latest-uv:
    name: Check that MkDocs can build the docs with the newest `uv`, no preinstalled Python
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
      - name: Install uv
        uses: astral-sh/setup-uv@2e657c127d5b1635d5a8e3fa40e0ac50a5bf6992
        with:
          version: latest
      - name: Check that `mkdocs` can build the docs
        run: uv run -- mkdocs build --strict
```

I can make this as a separate PR if you prefer.

Reply:
Will this be a separate job (in build.yml), or a separate workflow (.yml file)?

If it's a separate job -- it will block PRs on failure, and I'm not sure it's a good idea. I'd want to be alerted about `latest-uv` failures, but I wouldn't want to block all PRs until `latest-uv` is fixed.

If it's a separate workflow -- sure, no objections from me. Separate PR is best. I suppose we could even have a "docs" project (https://github.com/martinvonz/jj/projects?query=is%3Aopen) so that this isn't forgotten.

Reply:
I think it can be a separate job without being required. See this screenshot form a [random PR](https://github.com/martinvonz/jj/pull/4881):

<img width="886" alt="image" src="https://github.com/user-attachments/assets/e66e645c-6996-4139-9001-2559891f6ff1">

Reply:
Oh ok, found it in settings>rules>rulesets:

<img width="674" alt="image" src="https://github.com/user-attachments/assets/f78686ce-3e9a-4253-b0e9-70a5d73677c5">

I'll add `mkdocs-latest-uv` then

Reply:
> with the two Poetry jobs from build.yml (which get doubled to 4 for some reason)

FYI - the reason is that `push` runs on the contents of the branch, and `pull_request` runs on the contents of the simulated merge